core/src/main/scala-3/cats/instances/iarray.scala

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (c) 2015 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package cats
+package instances
+
+import cats.data.{Chain, Ior}
+import cats.instances.StaticMethods.{appendAll, appendAllMixed}
+import cats.kernel.compat.scalaVersionSpecific.*
+
+import scala.annotation.tailrec
+import scala.reflect.ClassTag
+import scala.collection.mutable.Builder
+
+trait IArrayInstances extends cats.kernel.instances.IArrayInstances {
+  implicit val catsStdInstancesForIArray
+    : Traverse[IArray] & Monad[IArray] & Alternative[IArray] & CoflatMap[IArray] & Align[IArray] =
+    new Traverse[IArray] with Monad[IArray] with Alternative[IArray] with CoflatMap[IArray] with Align[IArray] {
+
+      def empty[A]: IArray[A] = {
+        implicit val fakeClassTag: ClassTag[A] = summonFakeTag[A]
+        IArray.empty[A]
+      }
+
+      def combineK[A](x: IArray[A], y: IArray[A]): IArray[A] = {
```

Comment:
So, something like:
```scala
val cx = getClassTag(x)
val cy = getClassTag(y)
if (cx.runtimeClass.isAssignableFrom(cy)) {
  // cy <:< cx
  implicit val ctA: ClassTag[A] = cx.asInstanceOf[ClassTag[A]
  x.appendAll(y)
}
else if (/* check the opposite */ {
 
}
else {
  // neither may be subclasses because one could be a primitive Int and the other a boxed Int, for example.
  // just do boxing here
  implicit val fakeClassTag: ClassTag[A] = ClassTag.Object.asInstanceOf[ClassTag[A]]
  x.appendedAll(y)
}
```